truss/remote/baseten/api.py

```diff
@@ -0,0 +1,95 @@
+import logging
+
+import requests
+from truss.remote.baseten.auth import AuthToken, with_api_key
+from truss.remote.baseten.error import ApiError
+
+logger = logging.getLogger(__name__)
+
+
+def _post_graphql_query(auth_token: AuthToken, query_string: str) -> dict:
```

Comment:
I think it's better to make this a class and have `_post_graphql_query` pull both the configured URL and the headers from `self` instead of keeping these @with_api_key annotations, which are not helpful in this context anymore